packages/agent/src/agent/http/http.test.ts

```diff
@@ -253,3 +262,95 @@ describe('getDefaultFetch', () => {
     // TODO: Figure out how to test the self and default case errors
   });
 });
+
+describe('invalidate identity', () => {
+  const mockFetch: jest.Mock = jest.fn();
+  it('should allow its identity to be invalidated', () => {
+    const identity = new AnonymousIdentity();
+    const agent = new HttpAgent({ identity, fetch: mockFetch, host: 'http://localhost' });
+    const invalidate = () => agent.invalidateIdentity();
+    expect(invalidate).not.toThrowError();
+  });
+  it('should throw an error instead of making a call if its identity is invalidated', async () => {
+    const canisterId: Principal = Principal.fromText('2chl6-4hpzw-vqaaa-aaaaa-c');
+    const identity = new AnonymousIdentity();
+    const agent = new HttpAgent({ identity, fetch: mockFetch, host: 'http://localhost' });
+    agent.invalidateIdentity();
+
+    const expectedError =
+      "This identity has expired due this application's security policy. Please refresh your authentication.";
+
+    // Test Agent.call
+    await agent
+      .call(canisterId, {
+        methodName: 'test',
+        arg: new ArrayBuffer(16),
+      })
+      .catch((reason: AgentError) => {
```

Comment:
why not a `try / catch` block?

Reply:
you could also test that `mockFetch` has not been called

Reply:
The try catch block was interacting weirdly with this error in Jest. I can definitely check mockFetch though

Reply:
Yep, refactored with these updates